### PR TITLE
Web3 provider

### PR DESCRIPTION
**Pull request by fabriziovigevani**
This includes a basic web3 Provider. Still have to finish the contracts interaction that will be done in another PR. 

It also includes the CourtConfig provider and the graph endpoints based on the current chain.

Also fixed the header and sidebar.

### CODE REVIEW

File: src/queries/courtConfig.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import gql from 'graphql-tag'
```
Comment (rperez89) [inline src/queries/courtConfig.js]: I would rename this file to just court, as we are doing with the others queries and inside have the courtConfig query

File: src/providers/CourtConfig.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+const CHAIN_ID = environment('CHAIN_ID')
+const CourtConfigContext = React.createContext()
+
+function CourtConfigProvider({ children }) {
+  const courtAddress = networks[getNetworkName(CHAIN_ID)].court
+
+  console.log(courtAddress)
+
+  const [result] = useSubscription({
```
Comment (rperez89) [inline src/providers/CourtConfig.js]: i am using one hook per query so in the component we just use `data = useCourtSubscription`

File: src/queries/courtConfig.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+export const CourtConfig = gql`
```
Comment (rperez89) [inline src/queries/courtConfig.js]: one thing that facu mentioned in one of my pr's is to just query what we need not everything

File: src/providers/CourtConfig.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    query: CourtConfigSubscription,
+    variables: { id: courtAddress.toLocaleLowerCase() },
+  })
+
+  // TODO: handle possible errors
+  const courtConfig = result.data && result.data.courtConfig
+
+  return (
+    <CourtConfigContext.Provider value={courtConfig}>
```
Comment (rperez89) [inline src/providers/CourtConfig.js]: should we pass all the config or one prop by prop?

File: src/queries/courtConfig.js (hunk 3 above)
Comment (fabriziovigevani) [reply src/queries/courtConfig.js]: I think that eventually we will use all this data but will remove the ones we don't currently use and if needed let them be added later 👍

File: src/providers/CourtConfig.js (hunk 4 above)
Comment (fabriziovigevani) [reply src/providers/CourtConfig.js]: Unless we need to do some conversion i think that is ok to pass directly the configuration. Is there a reason that you think would be good to pass it like you mention?

File: package.json [hunk 5]
```diff
@@ -4,19 +4,32 @@
...
+    "subscriptions-transport-ws": "^0.9.16",
     "urql": "^1.6.3"
   },
   "scripts": {
-    "start": "react-app-rewired start",
-    "build": "react-app-rewired build",
+    "start": "npm run start:rpc",
+    "start:rpc": "REACT_APP_CHAIN_ID=1337 react-app-rewired start",
+    "start:mainnet": "REACT_APP_CHAIN_ID=1 react-app-rewired start",
+    "start:rinkeby": "REACT_APP_CHAIN_ID=3 react-app-rewired start",
```
Comment (bpierre) [inline package.json]:
```suggestion
    "start:rinkeby": "REACT_APP_CHAIN_ID=4 react-app-rewired start",
```

File: package.json [hunk 6]
```diff
@@ -4,19 +4,32 @@
...
+    "start:ropsten": "REACT_APP_CHAIN_ID=4 react-app-rewired start",
```
Comment (bpierre) [inline package.json]:
```suggestion
    "start:ropsten": "REACT_APP_CHAIN_ID=3 react-app-rewired start",
```

File: src/Header.js [hunk 7]
```diff
@@ -1,17 +1,58 @@
...
+        <div
+          css={`
+            margin-right: 12px;
+          `}
+        >
+          {account ? (
+            <span
+              css={`
+                background-color: #54c497;
+                color: white;
```
Comment (bpierre) [inline src/Header.js]: We should probably use the theme for these two.

File: src/web3-utils.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  if (chainId === '1') return 'mainnet'
+  if (chainId === '3') return 'ropsten'
+  if (chainId === '4') return 'rinkeby'
+
+  return DEFAULT_LOCAL_CHAIN
+}
+
+export function isLocalNetwork(chainId) {
+  return getNetworkName(chainId) === DEFAULT_LOCAL_CHAIN
+}
```
Comment (bpierre) [inline src/web3-utils.js]: What would you think about renaming this function `isLocalOrUnknownNetwork()`?

**Pull request merged**